samples/nrf_desktop/src/modules/ble_adv.c

```diff
@@ -343,9 +343,12 @@ static bool event_handler(const struct event_header *eh)
 			case STATE_GRACE_PERIOD:
 				err = ble_adv_stop();
 				if (!err) {
-					ble_adv_start();
+					err = ble_adv_start();
 				}
-				if (err) {
+				if (err == -ECONNREFUSED) {
```

Comment:
wouldn't it be better to catch it in `ble_adv_start`?

Reply:
I agree, moved inside `ble_adv_start`. But in such case I have to return 0 from `ble_adv_start`